src/alfasim_sdk/_internal/alfacase/plugin_alfacase_to_case.py

```diff
@@ -0,0 +1,404 @@
+from functools import lru_cache
+from pathlib import Path
+from pathlib import PurePosixPath
+from types import ModuleType
+from typing import Callable
+from typing import Dict
+from typing import List
+from typing import Optional
+from typing import Type
+from typing import Union
+
+import attr
+from barril.units import Array
+from barril.units import Scalar
+from typing_extensions import TypeGuard
+
+from alfasim_sdk import BaseField
+from alfasim_sdk._internal.alfacase.alfacase_to_case import DescriptionDocument
+from alfasim_sdk._internal.alfacase.alfacase_to_case import load_value
+from alfasim_sdk._internal.alfacase.alfacase_to_case import to_case_values
+from alfasim_sdk._internal.alfacase.case_description import CaseDescription
+from alfasim_sdk._internal.alfacase.case_description import PluginDescription
+from alfasim_sdk._internal.alfacase.case_description import PluginFileContent
+from alfasim_sdk._internal.alfacase.case_description import PluginInternalReference
+from alfasim_sdk._internal.alfacase.case_description import PluginMultipleReference
+from alfasim_sdk._internal.alfacase.case_description import PluginTableContainer
+from alfasim_sdk._internal.alfacase.case_description import PluginTracerReference
+from alfasim_sdk._internal.alfacase.case_description_attributes import (
+    InvalidPluginDataError,
+)
+from alfasim_sdk._internal.alfacase.plugin_introspection import get_attributes
+from alfasim_sdk._internal.types import Boolean
+from alfasim_sdk._internal.types import Enum
+from alfasim_sdk._internal.types import FileContent
+from alfasim_sdk._internal.types import MultipleReference
+from alfasim_sdk._internal.types import Quantity
+from alfasim_sdk._internal.types import Reference
+from alfasim_sdk._internal.types import String
+from alfasim_sdk._internal.types import Table
+from alfasim_sdk._internal.types import TracerType
+
+_CHILDREN_LIST_KEY = "_children_list"
+
+
+def get_unused_filename(alfacase_folder: Path, path: Path) -> Path:
+    """
+    Return a not used file name inside ``alfacase_folder``.
+    """
+    n = 1
+    extra = ""
+    while True:
+        fullpath = alfacase_folder / f"{path.stem}{extra}{path.suffix}"
+        if not fullpath.exists():
```

Comment:
I want to test that id do not exits (not a folder and not an file).
For negative tests I think the more generic `not path.exists()` a better choices or I will need to do `if not path.is_file() and not path.is_dir()``.

Reply:
Hmm but why do you care if a folder with same name exists, if you are trying to create a file?

But no problem, just pointing this out because I've seen code which wanted to check for files only, but still used `exists`... but if you really care about both, leave is as. 👍